src/utils/stack/stack.go

```diff
@@ -26,3 +26,11 @@ func (stack *Stack[K]) Pop() *K {
 func (stack *Stack[K]) Push(el K) {
 	*stack = append(*stack, el)
 }
+
+func (stack *Stack[K]) AsSlice() []K {
+	return *stack
+}
+
+func (stack *Stack[K]) Len() int {
+	return len(*stack)
+}
```

Comment:
I am not a fun of this additions I think method `Iter() []K` would be sufficient

Reply:
How would `Iter()` differ from `AsSlice()`?

Reply:
implemention wise its the same just something else when it coems to conception